reference/conanfile/attributes/sources.inc

```diff
@@ -111,4 +111,25 @@ For example, a *conandata.yml* with information about sources that looks like th
 .. code-block:: python
 
     def source(self):
-        tools.get(**self.conan_data["sources"][self.version])
\ No newline at end of file
+        tools.get(**self.conan_data["sources"][self.version])
+
+source_buildenv
+---------------
+
+Boolean attribute to opt-out from using tool requires in the `source()` method.
+
+Setting this attribute to `False` (default value `True`) will prevent Conan from inyecting the VirtualBuildEnv generated environment from tool requires
+when executing the `source()` method.
+
+.. code-block:: python
+   :emphasize-lines: 4,5,8
+
+    class MyConan:
+       name = "mylib"
+       version = "1.0.0"
+       source_buildenv = False
+       tool_requires = "7zip/1.2.0"
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
```

Comment:
Conan 1.X legacy code, please remember this is Conan 2.0, we don't even use ``tools.get()`` in ConanCenter in 1.X long time ago.

Reply:
```suggestion
    def source(self):
        get(self, **self.conan_data["sources"][self.version])
```